.github/workflows/ci.yml

```diff
@@ -112,6 +121,22 @@ jobs:
       - name: Test current JDBC vs old server
         if: always()
         run: $MAVEN test ${MAVEN_TEST} -pl :trino-test-jdbc-compatibility-old-server
+      - name: Upload test results
+        uses: actions/upload-artifact@v2
+        if: failure()
+        with:
+          name: result ${{ github.job }}
+          path: |
+            **/target/surefire-reports
+            **/target/checkstyle-*
+      - name: Upload test report
+        uses: actions/upload-artifact@v2
+        if: always()
```

Comment:
some reports are under `if: failure()` condition, some always. document each of them

Reply:
I added comments in every job.